src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/TemplateRenderer.cs

```diff
@@ -109,9 +109,14 @@ public IHtmlContent Render()
 
             foreach (string viewName in GetViewNames())
             {
-                var fullViewName = modeViewPath + "/" + viewName;
+                var viewEngineResult = _viewEngine.GetView(_viewContext.ExecutingFilePath, viewName, isMainPage: false);
+                if (!viewEngineResult.Success)
+                {
+                    // Success here is more common than with GetView() but GetView() is less expensive.
```

Comment:
Isn't this a pattern requirement and not a choice based on what's more expensive? We can't say `GetView` is necessarily less expensive for all `IViewEngine` implementations.